test/transformationsTest.js

```diff
@@ -0,0 +1,291 @@
+          F: 'female',
+          M: 'male',
+          O: 'other'
+        }
+
+        const result = transforms['mapCodes'](
+          inputCode,
+          null,
+          null,
+          null,
+          null,
+          functionParameters
+        )
+
+        t.equal(result, expectedOutput)
+        t.end()
+      }
+    )
+
+    t.test(
+      'should convert null codes to value if null case is specified',
+      t => {
+        const inputCode = null
+        const expectedOutput = 'unknown'
+
+        const functionParameters = {
+          F: 'female',
+          M: 'male',
+          O: 'other',
+          null: 'unknown'
+        }
+
+        const result = transforms['mapCodes'](
+          inputCode,
+          null,
+          null,
+          null,
+          null,
+          functionParameters
+        )
+
+        t.equal(result, expectedOutput)
+        t.end()
+      }
+    )
+
+    t.test(
+      'should convert null codes to default value if null case is not specified but default is',
+      t => {
+        const inputCode = null
+        const expectedOutput = 'unknown'
+
+        const functionParameters = {
+          F: 'female',
+          M: 'male',
+          O: 'other',
+          default: 'unknown'
+        }
+
+        const result = transforms['mapCodes'](
+          inputCode,
+          null,
+          null,
+          null,
+          null,
+          functionParameters
+        )
+
+        t.equal(result, expectedOutput)
+        t.end()
+      }
+    )
+
+    t.test('should convert null code to null if not handled', t => {
+      const inputCode = null
+      const expectedOutput = null
+
+      const functionParameters = {
+        F: 'female',
+        M: 'male',
+        O: 'other'
+      }
+
+      const result = transforms['mapCodes'](
+        inputCode,
+        null,
+        null,
+        null,
+        null,
+        functionParameters
+      )
+
+      t.equal(result, expectedOutput)
+      t.end()
+    })
+  })
+
+  t.test('booleanFlip()', t => {
+    t.test('should convert true to false', t => {
+      const booleanInput = true
+      const expectedOutput = false
+
+      const result = transforms.booleanFlip(booleanInput)
+
+      t.equal(result, expectedOutput)
+      t.end()
+    })
+
+    t.test('should convert false to true', t => {
+      const booleanInput = false
+      const expectedOutput = true
+
+      const result = transforms.booleanFlip(booleanInput)
+
+      t.equal(result, expectedOutput)
+      t.end()
+    })
+
+    t.test(
+      'should return null if input is not a primitive boolean data type',
+      t => {
+        const booleanInput = 'test'
+        const expectedOutput = null
+
+        const result = transforms.booleanFlip(booleanInput)
+
+        t.equal(result, expectedOutput)
+        t.end()
+      }
+    )
+
+    t.test('should return null if input is a falsy', t => {
+      const booleanInput = 0
+      const expectedOutput = null
```

Comment:
Would this not be returning `false` instead of `null`?
https://github.com/jembi/node-object-mapper/pull/1/files#diff-0dec99e818a2242edae67563675d4d06R25

Reply:
oh wait nvm, thats a different function call...